crates/router/src/connector/zen/transformers.rs

```diff
@@ -113,109 +138,254 @@ pub struct ZenItemObject {
+        })))
+    }
+}
 
-        let window_size = match (browser_info.screen_height, browser_info.screen_width) {
-            (250, 400) => "01",
-            (390, 400) => "02",
-            (500, 600) => "03",
-            (600, 400) => "04",
-            _ => "05",
-        }
-        .to_string();
-        let browser_details = ZenBrowserDetails {
-            color_depth: browser_info.color_depth.to_string(),
-            java_enabled: browser_info.java_enabled,
-            lang: browser_info.language,
-            screen_height: browser_info.screen_height.to_string(),
-            screen_width: browser_info.screen_width.to_string(),
-            timezone: browser_info.time_zone.to_string(),
-            accept_header: browser_info.accept_header,
-            window_size,
-            user_agent: browser_info.user_agent,
+impl TryFrom<(&types::PaymentsAuthorizeRouterData, &GooglePayWalletData)> for ZenPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (&types::PaymentsAuthorizeRouterData, &GooglePayWalletData),
+    ) -> Result<Self, Self::Error> {
+        let (item, gpay_pay_redirect_data) = value;
+        let amount = utils::to_currency_base_unit(item.request.amount, item.request.currency)?;
+        let browser_info = item.request.get_browser_info()?;
+        let browser_details = get_browser_details(&browser_info)?;
+        let ip = browser_info.get_ip_address()?;
+        let payment_specific_data = ZenPaymentData {
+            browser_details,
+            //Connector Specific for wallet
+            payment_type: ZenPaymentTypes::ExternalPaymentToken,
+            token: Some(gpay_pay_redirect_data.tokenization_data.token.clone()),
+            card: None,
+            descriptor: item.get_description()?.chars().take(24).collect(),
+            return_verify_url: item.request.router_return_url.clone(),
+        };
+        Ok(Self::ApiRequest(Box::new(ApiRequest {
+            merchant_transaction_id: item.attempt_id.clone(),
+            payment_channel: ZenPaymentChannels::PclGooglepay,
+            currency: item.request.currency,
+            payment_specific_data,
+            customer: get_customer(item, ip)?,
+            custom_ipn_url: item.request.get_webhook_url()?,
+            items: get_item_object(item, amount.clone())?,
+            amount,
+        })))
+    }
+}
+
+impl
+    TryFrom<(
+        &types::PaymentsAuthorizeRouterData,
+        &Box<ApplePayRedirectData>,
+    )> for ZenPaymentsRequest
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        value: (
+            &types::PaymentsAuthorizeRouterData,
+            &Box<ApplePayRedirectData>,
+        ),
+    ) -> Result<Self, Self::Error> {
+        let (item, _apple_pay_redirect_data) = value;
+        let amount = utils::to_currency_base_unit(item.request.amount, item.request.currency)?;
+        let connector_meta = item.get_connector_meta()?;
```

Comment:
how are we going to handle this in dashboard? is it already notified to them?

Reply:
Dashboard has to do the necessary changes, Already informed